generator/internal/discovery_resource_test.cc

```diff
@@ -0,0 +1,274 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "generator/internal/discovery_resource.h"
+#include "google/cloud/testing_util/status_matchers.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace generator_internal {
+namespace {
+
+using ::google::cloud::testing_util::StatusIs;
+using ::testing::Eq;
+using ::testing::HasSubstr;
+
+TEST(DiscoveryResourceTest, FormatRpcOptionsGetRegion) {
```

Comment:
I think there is missing test coverage because `FormatRpcOptions()` does several things at once. It sets:
- google.api.http
- google.api.method_signature
- googe.cloud.operation_service

I see in the implementation special handling for `put` / `patch` / `post`, but I don't see a test for the `put` case. Also, I only see a test for one of the operation_service thingies.

What I think you want to do is factor out the setting of each annotation, and unit test those individually. I think that will give us more focused unit tests and better coverage.